frontend/pkg/frontend/frontend.go

```diff
@@ -645,39 +647,41 @@ func (f *Frontend) ArmDeploymentPreflight(writer http.ResponseWriter, request *h
 	arm.WriteDeploymentPreflightResponse(writer, preflightErrors)
 }
 
-func logSubscriptionDifferences(logger *slog.Logger, oldSub, newSub *arm.Subscription) {
+func logSubscriptionDifferences(oldSub, newSub *arm.Subscription) []string {
```

Comment:
Last optional suggestion: change this function name to getSubscriptionsDifferences as we are no loner logging inside the function.